tests_e2e/tests/lib/virtual_machine_client.py

```diff
@@ -114,6 +117,23 @@ def restart(self, timeout: int = AzureClient._DEFAULT_TIMEOUT) -> None:
             operation_name=f"Restart {self._identifier}",
             timeout=timeout)
 
+        if not wait_for_boot:
+            return
+
+        start = datetime.datetime.now()
+        while datetime.datetime.now() < start + boot_timeout:
+            log.info("Waiting for VM %s to boot", self._identifier)
+            time.sleep(15)  # Note that we always sleep at least 1 time, to give the reboot time to start
+            instance_view = self.get_instance_view()
```

Comment:
does it throw an error if vm is unavailable during restart? It the expection is expected, we are not handling it here.